domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/SplitNameConstraintValidator.java

```diff
@@ -0,0 +1,27 @@
+
+package uk.gov.hmcts.cmc.domain.constraints;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.cmc.domain.models.otherparty.IndividualDetails;
+import uk.gov.hmcts.cmc.domain.models.otherparty.TheirDetails;
+import uk.gov.hmcts.cmc.domain.models.party.SplitNamedParty;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+// todo ROC-5160 delete this class once frontend is merged
+
+public class SplitNameConstraintValidator implements ConstraintValidator<SplitName, TheirDetails> {
+
+    @Override
+    public boolean isValid(TheirDetails theirDetails, ConstraintValidatorContext context) {
+        if (theirDetails instanceof SplitNamedParty) {
+            if (StringUtils.isBlank(theirDetails.getName()) &&
+                (StringUtils.isAnyBlank(((SplitNamedParty) theirDetails).getFirstName(), ((SplitNamedParty) theirDetails).getLastName()))) {
+                return false;
+            }
+            return true;
```

Comment:
Could directly return the negation of the `if` condition

Reply:
👍 same as above - will get rid of this whole class :D